pkg/argocd/update.go

```diff
@@ -243,6 +253,19 @@ func UpdateApplication(updateConf *UpdateConfiguration) ImageUpdaterResult {
 				result.NumImagesUpdated = 0
 			} else {
 				logCtx.Infof("Successfully updated the live application spec")
+				result.NumImagesUpdated += 1
+				if !updateConf.DisableKubeEvents && updateConf.KubeClient != nil {
+					annotations := map[string]string{}
+					for _, c := range changeList {
+						annotations[fmt.Sprintf("%s/oldtag", c.imageName)] = c.oldTag.TagName
+						annotations[fmt.Sprintf("%s/newtag", c.imageName)] = c.newTag.TagName
```

Comment:
On a second thought, I'm not sure whether this will work reliably. The name parts of annotations are rather strict, and must follow a rule:

```
regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]') with an optional DNS subdomain prefix and '/' (e.g. 'example.com/MyName')
```

Also, the name must be max 63 characters long. I think we must either make sure that the image name is properly normalized, look for another name for the annotation (and put the image name in its value) or think about a completely another solution (e.g. putting the details into the event's `Message`).

Is there a specific reason that you use an annotation for storing the updated images?

Reply:
Not really. I want just to "pass parameters" between the argocd-image-updater and a listener through events.

Reply:
New approach. Now annotations have less randomness and a fixed size around 40 characters

Reply:
Awesome! I tested it successfully, the event now looks like:

```
Name:             guestbook-multi.166fdad08646fd49
Namespace:        argocd
Labels:           <none>
Annotations:      argocd-image-updater.image-0/image-name: heptio-images/ks-guestbook-demo
                  argocd-image-updater.image-0/new-tag: 0.2
                  argocd-image-updater.image-0/old-tag: 0.1
                  argocd-image-updater.image-1/image-name: coredns/coredns
                  argocd-image-updater.image-1/new-tag: 1.8.3
                  argocd-image-updater.image-1/old-tag: 1.8.0
API Version:      v1               
Count:            1   
...
Involved Object:                
  API Version:       argoproj.io/v1alpha1
  Kind:              Application
  Name:              guestbook-multi                                                                     
  Namespace:         argocd                                                                                                                                                                                        
...
Message:             Successfully updated application 'guestbook-multi'
...
Source:
  Component:  ArgocdImageUpdater
Type:         Normal
```

I'm not sure whether it's intended to use the image name without the registry prefix (the Image Updater's API is possibly not good documented in that regard), but I'd suggest to change

```go
changeList = append(changeList, change{updateableImage.ImageName, updateableImage.ImageTag, containerImageNew.ImageTag})
```
to
```go
changeList = append(changeList, change{updateableImage.GetFullNameWithoutTag(), updateableImage.ImageTag, containerImageNew.ImageTag})
```

at the place where you store the image information to be propagated to the event. 

And I also guess this would be the last change required before merge :tada:

Reply:
I'm thinking in using annotations to compose a message on slack. So allowing the user to decide might  be a really cool idea :)

So, I finally decided to add both.

Reply:
I'm fine with that solution as well :+1: